File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minn=prices[0]
        profit=0
        for i in range(1,len(prices)):
            cost=prices[i]-minn
            profit=max(profit,cost)
            minn=min(minn,prices[i])
        return profit

        

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] arr) {
        int min=arr[0];
        int profit=0;
        int ans=0;
        for(int i=1;i<arr.length;i++){
            profit=arr[i]-min;
            ans=Math.max(ans,profit);
            if(arr[i]<min){
                min=arr[i];
            }
        }
        return ans;
    }
}